### PR TITLE
Add CI test runner with Github Actions

### LINKED ISSUE #512: [IDEA] run test suite in GitHub action
## Problem to solve

Set up a GitHub action that runs `yarn test`

### PR DESCRIPTION
**Pull request by seanmc9**
## What does this PR do and why?

This PR closes #512 by adding a Github Actions workflow that runs `yarn install` (with caching) and `yarn test` on commits and PRs to `main` to our CI pipeline.

## Screenshots or screen recordings

[This](https://github.com/seanmc9/juice-interface/runs/5142469887?check_suite_focus=true) is an example of a successful run.

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](../CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](../CONTRIBUTING.md#browser-support).

### CODE REVIEW

**Comment by seanmc9** [comment]
Only question I had with this is: Do we want CI to run when a PR is opened? When a PR is updated?

[These](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-object-34) are all of the available options.

File: .github/workflows/ci-run.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install with yarn
+        uses: borales/actions-yarn@v2.3.0
+        with:
+          cmd: install # will run `yarn install` command
```
[the remark this answers is not in the document]
Comment (seanmc9) [reply .github/workflows/ci-run.yml]: It also looks like[ borales is just not reccomended anymore](https://github.com/Borales/actions-yarn) so I just went with what you set up.

**Comment by seanmc9** [comment]
> This is great @seanmc9! I'd say let's run on on PR opening and updating as well as on any pushes to main. Though let's exclude draft PRs (maybe that means using the `ready_for_review` action?

Thanks, and for sure! The below covers PR opening and updating because it is actually just shorthand for the synchronize label [here](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-object-34).
```
  pull_request:
    branches:
      - main
```
And then to ignore draft PRs I just added `if: github.event.pull_request.draft == false` to the job.

**Comment by aeolianeth** [comment]
@seanmc9 thank you for working on this! Excited for this addition. Happy to approve, but I think you'll need to rebase with main before merging 💪

**Review by aeolianeth — APPROVED** [review]

**Comment by seanmc9** [comment]
> @seanmc9 thank you for working on this! Excited for this addition. Happy to approve, but I think you'll need to rebase with main before merging 💪

Sounds good and done! I think this may be blocked because @odd-amphora is a reviewer but whenever it is able to be merged I think just a squash merge works great - only real change has been to the .yml file.

**Pull request merged**